tests/neg-custom-args/captures/puretest.scala

```diff
@@ -0,0 +1,10 @@
+import caps.Pure
+def foo[C <: Pure]() = ()
+def bar[T, C <: Iterable[T] & Pure]() = ()
+def baz[CC[_] <: Pure]() = ()
+def bam[CC[A] <: Pure & Iterable[A]]() = ()
+def test =
+  foo[Int]()  // error
```

Comment:
```suggestion
def test =
  trait Foo extends Any with caps.Pure
  foo[Foo]()
```
Should we test it with some self-defined trait that extends caps.Pure?

Reply:
OK. Test was added.